lightpath/ui/gui.py

```diff
@@ -263,6 +271,13 @@ def update_path(self, *args,  **kwargs):
                 for widget in row:
                     widget.update_light(_in, _out)
 
+    def _destroy_lightpath_summary_signals(self, *args, **kwargs):
+        """ Update all widgets in rows """
+        # destroy all signals
+        logger.debug('destroying all lightpath_summary signals')
+        for row in self.rows:
+            row[0].device.lightpath_summary.destroy()
```

Comment:
Destroy is pretty heavy-handed as your PR text indicates; could just unsubscribing from subscriptions suffice? Do we need an atef-like [on-exit cleanup](https://github.com/pcdshub/atef/blob/a7c875f7f31c81a00eb174e43113866f05c682ec/atef/util.py#L19-L23)? (And heck, should that be in pcdsutils at this point?)

Reply:
You're right, I think `unsubscribe_all` should work.  `destroy()` sounds angry, but in this case is actually just unsubscribing from all the signals the `SummarySignal` watches.  

The atef cleanup reaches into the control layer and stops it directly, which is more widespread than I think we need here, since we know exactly where the callbacks are. It does seem like something that would be useful in general though, and it'd have been good to know that was an option.